src/Validation/URLValidationRESTController.php

```diff
@@ -0,0 +1,355 @@
+					'preview_nonce' => $preview_nonce,
+				],
+				$url
+			);
+		}
+
+		$this->url_validation_provider->get_url_validation( $url, get_post_type( $post ), true );
+		$validation_status_post = AMP_Validated_URL_Post_Type::get_invalid_url_post( $url );
+
+		$data = [
+			'results'     => [],
+			'review_link' => get_edit_post_link( $validation_status_post->ID, 'raw' ),
+		];
+
+		foreach ( AMP_Validated_URL_Post_Type::get_invalid_url_validation_errors( $validation_status_post ) as $result ) {
+			$data['results'][] = [
+				'error'       => $result['data'],
+				'forced'      => $result['forced'],
+				'sanitized'   => AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_ACK_ACCEPTED_STATUS === $result['status'],
+				'status'      => $result['status'],
+				'term_id'     => $result['term']->term_id,
+				'term_status' => $result['term_status'],
+				'title'       => AMP_Validation_Error_Taxonomy::get_error_title_from_code( $result['data'] ),
+			];
+		}
+
+		return rest_ensure_response( $this->filter_response_by_context( $data, $request['context'] ) );
+	}
+
+	/**
+	 * Retrieves the schema for plugin options provided by the endpoint.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->schema;
+		}
+
+		$sources_type = [
+			'items' => [
+				'context'    => [ self::CONTEXT_EDITOR ],
+				'properties' => [
+					'block_content_index' => [
+						'context' => [ self::CONTEXT_EDITOR ],
+						'type'    => 'integer',
+					],
+					'block_name'          => [
+						'context' => [ self::CONTEXT_EDITOR ],
+						'type'    => 'string',
+					],
+					'dependency_handle'   => [
+						'context' => [],
+						'type'    => 'string',
+					],
+					'dependency_type'     => [
+						'context' => [],
+						'type'    => 'string',
+					],
+					'extra_key'           => [
+						'context' => [],
+						'type'    => 'string',
+					],
+					'file'                => [
+						'context' => [],
+						'type'    => 'string',
+					],
+					'function'            => [
+						'context' => [],
+						'type'    => 'string',
+					],
+					'handle'              => [
+						'context' => [],
+						'type'    => 'string',
+					],
+					'hook'                => [
+						'context' => [],
+						'type'    => 'string',
+					],
+					'line'                => [
+						'context' => [],
+						'type'    => 'integer',
+					],
+					'name'                => [
+						'context' => [],
+						'type'    => 'string',
+					],
+					'post_id'             => [
+						'context' => [ self::CONTEXT_EDITOR ],
+						'type'    => 'integer',
+					],
+					'priority'            => [
+						'context' => [],
+						'type'    => 'integer',
+					],
+					'text'                => [
+						'context' => [],
+						'type'    => 'string',
+					],
+					'type'                => [
+						'context' => [],
+						'type'    => 'string',
+					],
+				],
+				'type'       => 'object',
+			],
+			'type'  => 'array',
```

Comment:
Note the type is `array` here, not `object`. So we should ensure that it actually returns an array.

Reply:
JSON array ensured in a91b14a.